src/windshaft/error.js

```diff
@@ -14,7 +14,7 @@ var WindshaftError = function (error) {
   }
   if (this.isAnalysisError(error.type)) {
     this.context = error.analysis && error.analysis.context;
-    this.analysisId = error.analysis && error.analysis.node_id;
+    this.analysisId = error.analysis && (error.analysis.node_id || error.analysis.id);
```

Comment:
I'm not sure if `error.analysis.id` will give us the id of the analysis that has some errors. If I recall correctly, this would be the id of the root node of the analysis. @dgaubert or @rochoa can't comment on this. It'd be nice if Maps API could categorise this error correctly (i.e: setting `error.analysis.node_id`) so we don't need to anything in cartodb.js.